src/polygon.jl

```diff
@@ -0,0 +1,69 @@
+"""
+    getlevel(dataset::ArchGDAL.AbstractDataset, reqlevel::Int) 
+Get layer of the desired level from the dataset
+"""
+function getlevel(dataset::ArchGDAL.AbstractDataset, reqlevel::Int)
+    # number of layers in the dataset
+    layers = ArchGDAL.nlayer(dataset)
+
+    for i = 0:layers-1
+        layer = ArchGDAL.getlayer(dataset, i)
+        layername = ArchGDAL.getname(layer)
+        layerlevel = split(layername, "_")[end]
+        if layerlevel == string(reqlevel)
+            return layer
+        end
+    end
+
+    throw(ArgumentError("valid levels for the given dataset are 0...$(layers-1)"))
+end
+
+"""
+    isgpkg(file)::Bool
+Returns true for .gpkg files
+"""
+function isgpkg(file)::Bool
+    filename, extension = splitext(file)
+    extension === ".gpkg"
+end
+
+"""
+    isvalidcode(code)::Bool
+Returns true for ISO3 country codes
+"""
+function isvalidcode(code)::Bool
+    # only allow ISO3 country codes like IND, USA, BRA
+    match(r"\b[A-Z]{3}\b", code) !== nothing
+end
+
+"""
+    get(code)
+Returns the MULTIPOLYGON data for the requested country.\n
+Input: ISO3 Country Code
+"""
+function get(code)
+
+    # only uppercase country codes are accepted
+    isvalidcode(code) || throw(ArgumentError("please provide standard ISO 3 country codes"))
+
+    dataid = "GADM_$code"
+
+    resource_data_path = download(dataid)
```

Comment:
We shouldn't be calling download everytime a call to `get` is performed. Instead we should assume that the dataset is available. If it is not available, then we ask the user if he/she wants to download the data with a YES/NO prompt.

Reply:
`download(dataid)` is our function that is being called from `download.jl`. 

DataDeps library does the same job as you described. It checks first if the dataset is available, if yes, it simply returns the path of the dataset. Data will not be downloaded every time. Also, DataDeps doesn't provide any specific function to check if dataset exists or not.

Reply:
What we need to do is to properly create a registry of the datasets downloaded. Right now we are relying too much on what DataDeps does. I would make sure that the download function in this package post-processes the data, and places it in a format that is easy to check later. Your post_fetch_method is currently just unpack, but I think we could unpack and do other things like rename the folder to the country code, and add an entry to a text file somewhere saying that the dataset is available. Let's first make sure that the other issues are fixed, and that the tests are passing. I will then try to install the package and experiment with it.

Reply:
We can discuss this more over our call soon.